Updated animist.js to use class for source links, fixing unique ID is…

biplab-sutradhar opened the pull request:
Updates animist.js to use classes for source links, resolving issue #145.

tdhock commented:
Hi, thanks for your contribution! 
Please also edit the test case, grep a_source_href in testthat/* files

biplab-sutradhar commented:
@tdhock I've edited the test case, thank you for your advice.

tdhock commented:
for PRs like this one that would close an issue, please put the following text in the first PR comment, which tells github to close the issue when the PR is merged.

Closes #145

tdhock commented:
I invited you to animint org, so after you accept that, you will have write permission to this repo.
please create future PRs using branches in this repo, instead of your fork.
(and please avoid pushing to master)

biplab-sutradhar commented:
Thank you, tdhock!

I've accepted the invitation to join the animint organization, and I’ll create future PRs using branches also make sure not to push directly to master.

biplab-sutradhar commented:
should I create a new branch here before commit

tdhock commented:
please just submit the changes to this branch/pr and then I will merge.

biplab-sutradhar commented:
@tdhock I have updated news.md, description and in this issue one error coming during testing "GitHub API error (403): Must have admin rights to the repository." This issue is most likely related to the environment variable.

tdhock commented:
that test failure always happens in PRs from branches in forks, because that test needs permission which is only available using token in this repo.

tdhock commented:
> I have updated news.md, description 

I don't see these updates, did you forget to push?

biplab-sutradhar commented:
@tdhock Sorry for my misunderstanding. Should I push my code to the pr branch, or to the bugfix/issue-123 branch (which I created for issue #154) If I create a new branch I need to make a new PR as well?

tdhock commented:
in this PR you should describe what changes you made in NEWS, and increment the version in DESCRIPTION
to update this PR you need to push to biplab-sutradhar:bugfix/issue-123

biplab-sutradhar commented:
Updated NEWS file to reflect changes made in resolving issue #154. 
 Added name and role as a contributor, incremented version number in DESCRIPTION based on the current date

biplab-sutradhar commented:
@tdhock extra newline issue was not showing in vscode hence I directly commit in github

tdhock commented:
please click "Files changed" tab at the top of this PR page, which leads to https://github.com/animint/animint2/pull/154/files which shows that there is still no change in DESCRIPTION and NEWS.
please make sure to preview your changed in Files changed tab before asking for review, because that is how other people review your PRs.

biplab-sutradhar commented:
@tdhock Sorry for the misunderstanding. I have updated.

tdhock commented on NEWS.md:
```diff
@@ -1,3 +1,7 @@
+# Changes in version 2024.10.10 (PR#145)
```
please change to PR#154 (145 is the issue)

biplab-sutradhar replied:
Yes, I changed to PR#154 on my last commit.

tdhock commented on NEWS.md:
```diff
@@ -1,3 +1,7 @@
+# Changes in version 2024.10.10 (PR#154)
+
+  - Updated `animist.js` to use classes for source links, resolving issue #154.
```
now the issue number is incorrect please revise

biplab-sutradhar replied:
https://github.com/animint/animint2/pull/154/commits/703d0656ad6b89d0f8ee52aaa175b2df7be8134c

this commit was correct, but I have updated to my last commit. Sorry for the misunderstanding.

tdhock commented:
ok thanks
